runtime/lua/vim/lsp/util.lua

```diff
@@ -2172,6 +2172,35 @@ function M.lookup_section(settings, section)
   return settings
 end
 
+--- Converts line range (0-based, end-inclusive) to lsp range,
+--- handles absence of a trailing newline
+---
+---@param bufnr integer
+---@param start_line integer
+---@param end_line integer
+---@param offset_encoding lsp.PositionEncodingKind
+---@return lsp.Range
+local function make_line_range_params(bufnr, start_line, end_line, offset_encoding)
```

Comment:
Related discussion https://github.com/neovim/neovim/pull/24499#issuecomment-1659006873

We need a `Position` and `Range` interface that can convert to/from the various indexing conventions.